src/c++/library/common.cc

```diff
@@ -232,6 +236,26 @@ InferInput::SetBinaryData(const bool binary_data)
   return Error::Success;
 }
 
+Error
+InferInput::ValidateData() const
+{
+  inference::DataType datatype =
+      triton::common::ProtocolStringToDataType(datatype_);
+  // String inputs will be checked at core and backend to reduce overhead.
+  if (datatype == inference::DataType::TYPE_STRING) {
+    return Error::Success;
+  }
+
+  int64_t expected_byte_size = triton::common::GetByteSize(datatype, shape_);
```

Comment:
I'm guessing having this in library/common is what introduces the protobuf dependency for HTTP client?

Reply:
Are there any concerns with introducing the new protobuf dependency to the HTTP client, or any alternatives? CC @GuanLuo @tanmayv25

Reply:
Correct